### PR TITLE
Changes as per Certificate Authentication

### PR DESCRIPTION
**Pull request by sarya-infoblox**
# Issue/Feature description

Certificate authentication is not supported.

# How it was fixed/implemented

Added Certificate and Key based authentication.

### CODE REVIEW

File: infoblox_client/connector.py [hunk 1]
```diff
@@ -109,10 +114,21 @@ def _parse_options(self, options):
...
-                msg = "WAPI config error. Option %s can not be blank" % attr
-                raise ib_ex.InfobloxConfigException(msg=msg)
+        def check(credentials):
+            for attr in credentials:
+                if not getattr(self, attr):
+                    msg = "WAPI config error. Option %s can not be blank" % attr
+                    raise ib_ex.InfobloxConfigException(msg=msg)
+
+        cerds_1 = {'host', 'username', 'password'}
+        cerds_2 = {'host', 'cert', 'key'}
```
Comment (achernevskii) [inline infoblox_client/connector.py:124]: Those var names are not descriptive. Can we rename those sets to something like `creds_userpass_auth` and `creds_cert_auth`.

File: infoblox_client/connector.py [hunk 2]
```diff
@@ -109,10 +114,21 @@ def _parse_options(self, options):
...
+        if cerds_1.issubset(options):
```
Comment (achernevskii) [inline infoblox_client/connector.py:125]: If both certificate and username/pass are provided, the client would prefer username and password. Is this behavior expected?

Comment (achernevskii) [reply infoblox_client/connector.py:125]: Aren't we running the same checks in here? Set method `issubset` will check if provided `options` set is subset of the creds_1. `Check` method will basically make the same check.

**Review by achernevskii — CHANGES_REQUESTED** [review]

**Comment by achernevskii** [comment]
Majority of the unit tests are failing, because [here](https://github.com/infobloxopen/infoblox-client/blob/Certificate_authentication/tests/test_connector.py#L59) `TestInfobloxConnector` class uses `mock.Mock` object to mock the input connector options. The mock object is not iterable, and we can't pass this object as argument to `set.issubset()` method.

I guess, we have to figure out some other way to mock the connector options, or we can replace `issubset` method with some other logic.

File: infoblox_client/connector.py [hunk 3]
```diff
@@ -75,7 +75,12 @@ class Connector(object):
                        'wapi_version': '2.7',
                        'max_results': None,
                        'log_api_calls_as_info': False,
-                       'paging': False}
+                       'paging': False,
+                       'username': None,
```
[the remark this answers is not in the document]
Comment (sarya-infoblox) [reply infoblox_client/connector.py:79]: Earlier when we used to make a connection to connector.py then values for 'username' and 'password' were passed by the user but now when we introduce new values 'cert' and 'key' then either pair values are passed which leads code to raise 'InfobloxConfigException' for remaining pair. 
In order to handle the exception, I declared the attributes 'username', 'password', 'cert', and 'key' as None (default values).

File: infoblox_client/connector.py (hunk 1 above)
Comment (sarya-infoblox) [reply infoblox_client/connector.py:124]: I will modify the names as per your suggestion.

File: infoblox_client/connector.py (hunk 2 above)
Comment (sarya-infoblox) [reply infoblox_client/connector.py:125]: yes, if both values are provided it will prefer username and password.

**Review by ranjishmp — APPROVED** [review]

**Review by achernevskii — APPROVED** [review]

**Pull request merged**